nucypher/crypto/powers.py

```diff
@@ -108,7 +108,7 @@ class AccountLocked(PowerUpError):
         """Raised when signing cannot be performed due to a locked account"""
         pass
 
-    @validate_checksum_address
+    # @validate_checksum_address  # TODO: Workaround when `account is NO_WORKER_BONDED` raising invalid checksum address.
```

Comment:
Interesting 🤔  - what's cas when a TransactionPower is attempted to be instantiated without a valid account address?

Reply:
When `account is NO_WORKER_BONDED` which is somewhat of a corner case.

Reply:
I was wondering more along the lines of, why not prevent TransactionPower from being initialized in that case? - initializing it with the account not being set to an actual address does not make sense to me.

Reply:
Unfortunately, I do not have a solid answer for you at this time.  This test suddenly started failing for this seemingly unrelated reason, so commenting out this line and leaving a comment is in an effort to make the smallest change possible while for PR precision.

To address your suggestion - I agree - There can be some higher-level handling preventing the `TransactingPower` from being set if there is  no address.  One issue that needs to faced is that `NO_WORKER_ADDRESS` may be returned from `Ursula.worker_address` which results in mixed signaling and property logic.

Reply:
There are still two failing tests caused by this - open to suggestions on how to get CI to accept the pre-push changes despite this diversion.

Reply:
Are the two failing tests the following?
1. tests/acceptance/cli/ursula/test_stakeholder_and_ursula.py::test_stake_unbond_worker
2. tests/acceptance/blockchain/actors/test_staker.py::test_staker_manages_winding_down 

For 1) it may just be a "Falsey" issue
```
E       AssertionError: assert not NO_WORKER_BONDED (0x0000000000000000000000000000000000000000)
 
E        +  where NO_WORKER_BONDED (0x0000000000000000000000000000000000000000) = Staker(address='0xaaa23A5c74aBA6ca5E7c09337d5317A7C4563075').worker_address
```

 I added the following  to the test:
```
print(f">>>> Derek {staker.worker_address}")
```
and obtained `>>>> Derek 0x0000000000000000000000000000000000000000`

For 2) it seems that the test assumes that the staker already has a bonded worker, but I don't think it does. You may have to add something like the following before using `staker.worker_address`
```
    worker_address = testerchain.unassigned_accounts[-1] <--- I saw this in other tests
    staker.bond_worker(worker_address=worker_address)
```

Reply:
For an accurate representation of this draft - do not consider the last commit (which is me hacking and slashing the other day looking for a solution.)

I dropped 2e72d373592708b666326b3f634aa5e33e39ebc3. Take a peek at CI afterwards to get the same initial impression I had after finished the pre-push changes.

Reply:
Okay - this is a better starting place for examination: <https://app.circleci.com/pipelines/github/nucypher/nucypher/5057/workflows/0108b710-9cf0-4587-afc8-27136231029d/jobs/91189>

Reply:
:heavy_check_mark: